typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+	return [apiServiceContainerCreationConfig, apiServiceGenerateRunConfigFunc];
+}
+
+async function getApiServiceConfigInitializingFunc(datastoreClient: DatastoreClient): Promise<(fp: number) => Promise<Result<null, Error>>> { //Note: Making simplification that file descriptor is just number
+	const configInitializingFunc: (fp: number) => Promise<Result<null, Error>> = async (fp: number) => { //TOOD (Ali) - might require changes in ConfigRunFactory in kurt-client due to async
+		log.debug("Datastore IP: "+datastoreClient.getIpAddr+" , port: "+datastoreClient.getPort+"");
+		const configObj: DatastoreConfig = new DatastoreConfig(datastoreClient.getIpAddr(), datastoreClient.getPort());
+		let configBytes: string;
+		try { 
+			configBytes = JSON.stringify(configObj);
+		} catch(jsonErr) {
+			return err(jsonErr);
+		}
+
+		log.debug("API config JSON: " + String(configBytes));
+
+
+		const writeFilePromise: Promise<ResultAsync<null, Error>> = new Promise((resolve, _unusedReject) => {
+			fs.writeFile(fp, configBytes, (error: Error) => {
+				if (error === null) {
+					resolve(okAsync(null));
+				} else {
+					resolve(errAsync(error));
+				}
+			})
+		});
+		const writeFileResult: Result<fs.Stats, Error> = await writeFilePromise;
+		if (!writeFileResult.isOk()) {
+			return err(writeFileResult.error);
+		}
+	
+		return ok(null);
+	}
+	return configInitializingFunc;
+}
+
+async function getApiServiceContainerCreationConfig(configInitializingFunc: (fp: number) => Promise<Result<null, Error>>): Promise<ContainerCreationConfig> {
+	const apiServiceContainerCreationConfig: ContainerCreationConfig = new ContainerCreationConfigBuilder(
+		API_SERVICE_IMAGE,
+	).withUsedPorts(
+		new Set(API_SERVICE_PORT+"/tcp")
+	).withGeneratedFiles(new Map().set(
+		CONFIG_FILE_KEY, configInitializingFunc //TODO (Ali) - might need to wrap value in kurt client of this func inside a promise
+	)).build();
+	return apiServiceContainerCreationConfig;
+}
+
+async function getApiServiceRunConfigFunc(): Promise<(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>> {
+	const apiServiceRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = 
+	(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => {
+		if (!generatedFileFilepaths.has(CONFIG_FILE_KEY)) {
+			return err(new Error("No filepath found for config file key '"+ CONFIG_FILE_KEY +"'"));
+		}
+		const configFilepath: string = generatedFileFilepaths[CONFIG_FILE_KEY];
+		const startCmd: string[] = [
+			"./api.bin",
+			"--config",
+			configFilepath
+		]
+
+		const result: ContainerCreationConfig = new ContainerRunConfigBuilder().withCmdOverride(startCmd).build();
```

Comment:
these type assertions don't match - `result: ContainerCreationConfig`, but `ContainerRunConfigBuilder().....build()`. Does this compile?